src/core/processing/models/qgsprocessingmodeloutput.cpp

```diff
@@ -28,17 +28,52 @@ QVariant QgsProcessingModelOutput::toVariant() const
 {
   QVariantMap map;
   map.insert( QStringLiteral( "name" ), mName );
+
+  if ( mDefaultValue.canConvert<QgsProcessingOutputLayerDefinition>() )
+  {
+    QVariantMap defaultMap = mDefaultValue.value<QgsProcessingOutputLayerDefinition>().toVariant().toMap();
+    defaultMap.insert( QStringLiteral( "class" ), QStringLiteral( "QgsProcessingOutputLayerDefinition" ) );
+    map.insert( QStringLiteral( "default_value" ), defaultMap );
+  }
+  else
+  {
+    map.insert( QStringLiteral( "default_value" ), mDefaultValue );
+  }
+
   map.insert( QStringLiteral( "child_id" ), mChildId );
   map.insert( QStringLiteral( "output_name" ), mOutputName );
+  map.insert( QStringLiteral( "mandatory" ), mMandatory );
   saveCommonProperties( map );
   return map;
 }
 
 bool QgsProcessingModelOutput::loadVariant( const QVariantMap &map )
 {
   mName = map.value( QStringLiteral( "name" ) ).toString();
+
+  QVariant defaultValue = map.value( QStringLiteral( "default_value" ) );
+  if ( defaultValue.type() == QVariant::Map )
+  {
+    QVariantMap defaultMap = defaultValue.toMap();
+    if ( defaultMap["class"] == QStringLiteral( "QgsProcessingOutputLayerDefinition" ) )
+    {
+      QgsProcessingOutputLayerDefinition value( QStringLiteral( "" ) );
```

Comment:
QStringLiteral( "" ) -> QString()

Reply:
I've tried, but I get an error I do not understand: 
```
/home/amorvan/dev/QGIS/src/core/processing/models/qgsprocessingmodeloutput.cpp: In member function ‘bool QgsProcessingModelOutput::loadVariant(const QVariantMap&)’:
/home/amorvan/dev/QGIS/src/core/processing/models/qgsprocessingmodeloutput.cpp:61:13: error: request for member ‘loadVariant’ in ‘value’, which is of non-class type ‘QgsProcessingOutputLayerDefinition(QString (*)())’
       value.loadVariant( defaultMap );
             ^
/home/amorvan/dev/QGIS/src/core/processing/models/qgsprocessingmodeloutput.cpp:62:39: warning: the address of ‘QgsProcessingOutputLayerDefinition value(QString (*)())’ will always evaluate as ‘true’ [-Waddress]
       mDefaultValue = QVariant( value );
                                       ^
[156/464] Building CXX object src/core/CMakeFiles/qgis_core.dir/processing/models/qgsprocessingmodelalgorithm.cpp.o
ninja: build stopped: subcommand failed.
```

Reply:
Got it with `QgsProcessingOutputLayerDefinition value;`, as QString() is the default value.